internal/core/command/router.go

```diff
@@ -7,41 +7,41 @@
 package command
 
 import (
-	"net/http"
+	commandController "github.com/edgexfoundry/edgex-go/internal/core/command/controller/http"
+	commonController "github.com/edgexfoundry/edgex-go/internal/pkg/controller/http"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
 
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/container"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/handlers"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/di"
 	"github.com/edgexfoundry/go-mod-core-contracts/v3/common"
-	"github.com/gorilla/mux"
 
-	commandController "github.com/edgexfoundry/edgex-go/internal/core/command/controller/http"
-	commonController "github.com/edgexfoundry/edgex-go/internal/pkg/controller/http"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
+	"github.com/labstack/echo/v4"
 )
 
-func LoadRestRoutes(r *mux.Router, dic *di.Container, serviceName string) {
-	// r.UseEncodedPath() tells the router to match the encoded original path to the routes
-	r.UseEncodedPath()
-
+func LoadRestRoutes(r *echo.Echo, dic *di.Container, serviceName string) {
 	lc := container.LoggingClientFrom(dic.Get)
 	secretProvider := container.SecretProviderExtFrom(dic.Get)
 	authenticationHook := handlers.AutoConfigAuthenticationFunc(secretProvider, lc)
 
 	// Common
 	cc := commonController.NewCommonController(dic, serviceName)
-	r.HandleFunc(common.ApiPingRoute, cc.Ping).Methods(http.MethodGet) // Health check is always unauthenticated
-	r.HandleFunc(common.ApiVersionRoute, authenticationHook(cc.Version)).Methods(http.MethodGet)
-	r.HandleFunc(common.ApiConfigRoute, authenticationHook(cc.Config)).Methods(http.MethodGet)
+
+	g := r.Group(common.ApiBase)
+	g.Use(authenticationHook)
+	g.Use(correlation.ManageHeader)
+	g.Use(correlation.LoggingMiddleware(container.LoggingClientFrom(dic.Get)))
+	g.Use(correlation.UrlDecodeMiddleware(container.LoggingClientFrom(dic.Get)))
+
+	g.GET("/ping", cc.Ping) // Health check is always unauthenticated
```

Comment:
The `ping` should not apply the `authenticationHook` API, let's just ignore this in this PR. The logic will be implemented in `go-mod-bootstrap` as all the common APIs will be moved to it.

Reply:
The consequence of authenticating the /ping route is that Consul will mark any such services as failed.

Reply:
Agreed that making Ping always unauthenticated can be handled in the new Common Controller code in go-mod-bootstrap. That code was just merged, so this [PR ](https://github.com/edgexfoundry/go-mod-bootstrap/pull/557) needs to be update to include that code.